api/src/importers/cors/datasource.js

```diff
@@ -0,0 +1,149 @@
+const defaultLog = require('../../utils/logger')('cors-csv-datasource');
+const RECORD_TYPE = require('../../utils/constants/record-type-enum');
+
+class CorsCsvDataSource {
+  /**
+   * Creates an instance of DataSource.
+   *
+   * @param {*} taskAuditRecord audit record hook for this import instance
+   * @param {*} auth_payload information about the user account that started this update
+   * @param {*} recordType record type to create from the csv file
+   * @param {*} csvRows csv file to import
+   * @memberof CorsCsvDataSource
+   */
+  constructor(taskAuditRecord, auth_payload, recordType, csvRows) {
+    this.taskAuditRecord = taskAuditRecord;
+    this.auth_payload = auth_payload;
+    this.recordType = recordType;
+    this.csvRows = csvRows;
+
+    // Set initial status
+    this.status = { itemsProcessed: 0, itemTotal: 0, individualRecordStatus: [] };
+  }
+
+  /**
+   * Run the Cors csv importer.
+   *
+   * @returns
+   * @memberof CorsCsvDataSource
+   */
+  async run() {
+    defaultLog.info('run - import cors-csv');
+    await this.taskAuditRecord.updateTaskRecord({ status: 'Running', itemTotal: this.csvRows.length });
+
+    return await this.batchProcessRecords();
+  }
+
+  /**
+   * Runs processRecord() on each csv row, in batches.
+   *
+   * Batch size configured by env variable `CSV_IMPORT_BATCH_SIZE` if it exists, or 100 by default.
+   *
+   * @memberof CorsCsvDataSource
+   */
+  async batchProcessRecords() {
+    try {
+      let batchSize = process.env.CSV_IMPORT_BATCH_SIZE || 100;
+
+      const recordTypeConfig = this.getRecordTypeConfig();
+
+      if (!recordTypeConfig) {
+        throw Error('batchProcessRecords - failed to find matching recordTypeConfig.');
+      }
+
+      let promises = [];
+      for (let i = 0; i < this.csvRows.length; i++) {
+        promises.push(this.processRecord(this.csvRows[i], recordTypeConfig));
+
+        if (i % batchSize === 0 || i === this.csvRows.length - 1) {
+          await Promise.all(promises);
+          promises = [];
+        }
+      }
```

Comment:
It'd be nice here to updateTaskRecord to get a bit more granularity on the number of items being processed.

Reply:
The audit record gets updated in processRecords, once for each csvRow.  So it should already be updating as much as possible.